### PR TITLE
PLUGIN-574: Updated documentation of Salesforce plugins with entities that are not supported by bulk API

### PR DESCRIPTION
**Pull request by flakrimjusufi**
Salesforce entities not supported by plugins
JIRA Ticket: https://cdap.atlassian.net/browse/PLUGIN-574

### CODE REVIEW

File: docs/Salesforce-streamingsource.md [hunk 1]
```diff
@@ -69,6 +69,46 @@ for the evaluated records only if they match the criteria specified in the WHERE
 Salesforce and generate SOQL query (`select <FIELD_1, FIELD_2, ..., FIELD_N> from ${sObjectName}`).
 Ignored if SOQL query is provided.
 
+There are also **sObjects** that are not supported in the Bulk API of Salesforce.
+When a job is created using an object that is not supported in the Bulk API, "_Entity is not supported by the Bulk API_" is thrown.
```
Comment (CuriousVini) [inline docs/Salesforce-streamingsource.md]: when would plugin use build api? How would plugin user know it?

Comment (CuriousVini) [reply docs/Salesforce-streamingsource.md]: are we using bulk api in streaming source?

Comment (flakrimjusufi) [reply docs/Salesforce-streamingsource.md]: Thanks for catching this. Streaming source does not use Bulk API. I removed it from the documentation of Salesfoce-streamingsource in the last update.

File: docs/Salesforce-batchsink.md [hunk 2]
```diff
@@ -30,6 +30,46 @@ A Salesforce connected application must be created in order to get a client secr
 
 **SObject Name:** Salesforce object name to insert records into.
 
+There are also **sObjects** that are not supported in the Bulk API of Salesforce.
+When a job is created using an object that is not supported in the Bulk API, "_Entity is not supported by the Bulk API_" is thrown.
```
Comment (CuriousVini) [inline docs/Salesforce-batchsink.md:34]: do we always use bulk api for source and sink? Are there any cases when we use non-bulk api?

File: docs/SalesforceMultiObjects-batchsource.md [hunk 3]
```diff
@@ -158,3 +158,43 @@ multisink.Activity =
     ]
   }
 ```
+
+There are also **sObjects** that are not supported in the Bulk API of Salesforce.
+When a job is created using an object that is not supported in the Bulk API, "_Entity is not supported by the Bulk API_" is thrown.
```
Comment (CuriousVini) [inline docs/SalesforceMultiObjects-batchsource.md:163]: does multi object source also use bulk api?

File: docs/Salesforce-batchsink.md (hunk 2 above)
Comment (flakrimjusufi) [reply docs/Salesforce-batchsink.md:34]: There's one case when we don't use bulk API in source and sink. 

According to java doc in **SalesforceInputFormat.java**: 

```
Based on query length sends query to Salesforce to receive array of batch info.
If query is within limit, executes original query. If not, switches to wide object logic,
i.e. generates Id query to retrieve batch info for Ids only that will be used later
to retrieve data using SOAP API.
```

https://github.com/data-integrations/salesforce/blob/48270f0815e85f1f62b7cf73c7acf4a80a6563fe/src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceInputFormat.java#L111-L133

File: docs/SalesforceMultiObjects-batchsource.md (hunk 3 above)
Comment (flakrimjusufi) [reply docs/SalesforceMultiObjects-batchsource.md:163]: SalesForceBatchMultiSource uses SalesforceInputFormatProvider which uses Bulk Api. 

https://github.com/data-integrations/salesforce/blob/48270f0815e85f1f62b7cf73c7acf4a80a6563fe/src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java#L84-L85

File: docs/Salesforce-batchsink.md (hunk 2 above)
Comment (CuriousVini) [reply docs/Salesforce-batchsink.md:34]: I see. In that case, Could you please make sure documentation reflects those cases.

File: docs/SalesforceMultiObjects-batchsource.md (hunk 3 above)
Comment (CuriousVini) [reply docs/SalesforceMultiObjects-batchsource.md:163]: I see. sounds good

File: docs/Salesforce-batchsink.md (hunk 2 above)
Comment (flakrimjusufi) [reply docs/Salesforce-batchsink.md:34]: > I see. In that case, Could you please make sure documentation reflects those cases.

PR updated accordingly.

**Review by rmstar — APPROVED** [review]

File: docs/SalesforceMultiObjects-batchsource.md [hunk 4]
```diff
@@ -158,3 +158,43 @@ multisink.Activity =
     ]
   }
 ```
```
Comment (rmstar) [inline docs/SalesforceMultiObjects-batchsource.md:160]: Is the `Cases when BULK API is not used` section also applicable to multi object source? If yes, can you please add it here?

Comment (flakrimjusufi) [reply docs/SalesforceMultiObjects-batchsource.md:160]: Yes, this section is also applicable to multi object source and it's already added in the documentation.

**Pull request merged**